include/RED4ext/RedProcess.hpp

```diff
@@ -0,0 +1,78 @@
+#include <RED4ext/RED4ext.hpp>
+
+namespace RED4ext::red
+{
+
+struct Process
+{
+    struct FixedWString
+    {
+        uint32_t length;
+        uint32_t maxLength;
+        const wchar_t* str;
+    };
+
+    enum class ExecutionFlags : unsigned char
+    {
+        // unsets CREATE_NO_WINDOW
+        ShouldCreateWindow = 0x1,
+        // sets CREATE_BREAKAWAY_FROM_JOB | CREATE_SUSPENDED in Process Creation Flags
+        BreakawayAndSuspend = 0x2,
+        // unsets bInheritHandles
+        NoInheritHandles = 0x4
+    };
+
+    enum class ReadFlags : unsigned char
+    {
+        // open & read regardless of other options
+        Unk1 = 0x1,
+        // opens file from name provided and writes output
+        WriteToFile = 0x2,
+        // writes output to CString
+        GetCString = 0x4
+    };
+
+    static const constexpr uint32_t defaultTimeout = 60000;
+
+    // creates the read & write pipes, and handle information
```

Comment:
If this needs to be documented I would do it using `doxygen`, by default VS generates one if you type `/**` (or depending on the setting you have set).